pkg/controller/registrysyncer/registrysyncer.go

```diff
@@ -151,14 +151,36 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, log *
 	if n := len(imageStreamNameAndTag); n != 2 {
 		return fmt.Errorf("when splitting imagestreamtagname %s by : expected two results, got %d", req.Name, n)
 	}
-	imageStreamName, imageTag := imageStreamNameAndTag[0], imageStreamNameAndTag[1]
+	imageStreamName := imageStreamNameAndTag[0]
 	isName := types.NamespacedName{Namespace: req.Namespace, Name: imageStreamName}
 	sourceImageStream := &imagev1.ImageStream{}
 	if err := r.registryClients[srcClusterName].Get(ctx, isName, sourceImageStream); err != nil {
 		// received a request on the isTag, but the 'is' is no longer there
 		return fmt.Errorf("failed to get imageStream %s from cluster %s: %w", isName.String(), srcClusterName, err)
 	}
 
+	// if found soft-delete annotation on any IS, delete the IS on all clusters
```

Comment:
If the delete failed for any reason: sync (reconcile), annotation (by release-controller), and then delete (reconcile).